### PR TITLE
SourceLink support

### PR DESCRIPTION
**Pull request by ge0rdi**
This PR adds support for SourceLink so that debuggers can obtain proper version of source files automatically.
It uses official `Microsoft.SourceLink.GitHub` NuGet package that does all the work and is referenced by all projects.

Additionally, it fixes paths stored in PDBs so that they are all relative to repo root.
Currently these paths are solution-relative so it is not possible to map them to repo paths for some binaries (`systeminformer.sys`, plugins, ...).

### CODE REVIEW

File: tools/CustomBuildTool/bin/Release/CustomBuildTool.exe
Comment (ge0rdi) [inline tools/CustomBuildTool/bin/Release/CustomBuildTool.exe:1]: I have built custom build tool using `build_tools.cmd`. But maybe you'd like to rebuild it on your own?

**Comment by ge0rdi** [comment]
It will be necessary to also rebuild `thirdparty.lib` because current one contains source paths relative to `tools/thirdparty` (such as `pcre/pcre2_compile.c`) and these don't match SourceLink data.

**Comment by jxy-s** [comment]
Just finished testing, looks to work as advertised, nice work! 👍

File: tools/CustomBuildTool/bin/Release/CustomBuildTool.exe
Comment (jxy-s) [reply tools/CustomBuildTool/bin/Release/CustomBuildTool.exe:1]: I'll take care of this post-merge.

**Review by jxy-s — APPROVED** [review]

**Pull request merged**